src/Grpc.Net.ClientFactory/Internal/GrpcCallInvokerFactory.cs

```diff
@@ -75,12 +77,58 @@ private CallInvoker CreateInvoker(EntryKey key)
         try
         {
             var httpClientFactoryOptions = _httpClientFactoryOptionsMonitor.Get(name);
+            var clientFactoryOptions = _grpcClientFactoryOptionsMonitor.Get(name);
+
+            // gRPC channel is configured with a handler instead of a client, so HttpClientActions aren't used directly.
+            // To capture HttpClient configuration, a temp HttpClient is created and configured using HttpClientActions.
+            // Values from the temp HttpClient are then copied to the gRPC channel.
+            // Only values with overlap on both types are copied so log a message about the limitations.
             if (httpClientFactoryOptions.HttpClientActions.Count > 0)
             {
-                Log.UnsupportedHttpClientActions(_logger, name);
+                Log.HttpClientActionsPartiallySupported(_logger, name);
+
+                var httpClient = new HttpClient(NullHttpMessageHandler.Instance);
+                foreach (var applyOptions in httpClientFactoryOptions.HttpClientActions)
+                {
+                    applyOptions(httpClient);
+                }
+
+                // Copy configuration from HttpClient to GrpcChannel/CallOptions.
+                // This configuration should be overriden by gRPC specific config methods.
+                if (clientFactoryOptions.Address == null)
+                {
+                    clientFactoryOptions.Address = httpClient.BaseAddress;
```

Comment:
Would it be interesting to log when the HttpClient value is present but ignored?

Reply:
I don't think we need to go to this detail.